xdsl/irdl/constraints.py

```diff
@@ -48,17 +54,35 @@ def get_range_variable(self, key: str) -> tuple[Attribute, ...] | None:
     def get_int_variable(self, key: str) -> int | None:
         return self._int_variables.get(key)
 
-    def set_variable(self, key: str, attr: Attribute):
+    def set_attr_variable(self, key: str, attr: Attribute):
+        if key in self._variables and self._variables[key] != attr:
+            raise ValueError(
+                f"Variable {key} was set to different attributes:\n"
+                f"Possible values: {self._variables[key]}, {attr}"
+            )
```

Comment:
This isn't being used anymore, either I can revert it or we could keep this logic (and change it to a verify error) and simplify the logic in `VarConstraint`. Any thoughts?

Reply:
I think I wrote this in zulip, but the issue with moving all the logic here is that we would end up reverifying the constraint attached to `VarConstraint` each time the variable was encountered. I'm leaning towards keeping the logic in `VarConstraint` and verify, and reverting the changes to `ConstraintContext`